test/libsolidity/syntaxTests/viewPureChecker/inline_assembly_instructions_allowed_pure.sol

```diff
@@ -0,0 +1,90 @@
+contract C {
+    function f() public pure {
+        assembly {
+            stop()
+            pop(add(0, 1))
+            pop(sub(0, 1))
+            pop(mul(0, 1))
+            pop(div(0, 1))
+            pop(sdiv(0, 1))
+            pop(mod(0, 1))
+            pop(smod(0, 1))
+            pop(exp(0, 1))
+            pop(not(0))
+            pop(lt(0, 1))
+            pop(gt(0, 1))
+            pop(slt(0, 1))
+            pop(sgt(0, 1))
+            pop(eq(0, 1))
+            pop(iszero(0))
+            pop(and(0, 1))
+            pop(or(0, 1))
+            pop(xor(0, 1))
+            pop(byte(0, 1))
+            pop(shl(0, 1))
+            pop(shr(0, 1))
+            pop(sar(0, 1))
+            pop(addmod(0, 1, 2))
+            pop(mulmod(0, 1, 2))
+            pop(signextend(0, 1))
+            pop(keccak256(0, 1))
+            pop(0)
+            pop(mload(0))
+            mstore(0, 1)
+            mstore8(0, 1)
+            //pop(sload(0))
+            //sstore(0, 1)
+            //pop(gas())
```

Comment:
I did not check the actual current behaviour. But wouldn't it be better to not comment those out, so that we can see the error messages later?

Reply:
Yeah, that's why I also added `inline_assembly_instructions_disallowed_pure.sol`. It contains only those commented-out opcodes.

I left them commented-out here because otherwise it's not that easy to see which ones are missing from the list.

Also, I want this case to pass without error because then it goes all the way through codegen. The failing one (obviously) stops at the analysis stage.

Reply:
not uncomment, but let the errors there.. e.g.
```
// ====
// EVMVersion: >=london
// ----
// Warning 5740: (94-1721): Unreachable code.
...
// TypeError 8961: (1759-1774): Function cannot be declared as pure because this expression (potentially) modifies the state.
// TypeError 8961: (1809-1819): Function cannot be declared as pure because this expression (potentially) modifies the state.
...
// TypeError 8961: (1919-1941): Function cannot be declared as pure because this expression (potentially) modifies the state.
```

Reply:
not uncomment, but let the errors there.. e.g.
```
// ====
// EVMVersion: >=london
// ----
// Warning 5740: (94-1721): Unreachable code.
...
// TypeError 8961: (1759-1774): Function cannot be declared as pure because this expression (potentially) modifies the state.
// TypeError 8961: (1809-1819): Function cannot be declared as pure because this expression (potentially) modifies the state.
...
// TypeError 8961: (1919-1941): Function cannot be declared as pure because this expression (potentially) modifies the state.
```

Reply:
Right now I would prefer having those errors.. but I also find the error messages a bit poor.. I think it would be better to know exactly what expression actually caused it

Reply:
Maybe both version would make sense.. one where the stuff is commented out, and one that will produce an error?

Reply:
I don't understand. If you don't uncomment them, how would you get the error messages?

Reply:
Ah sorry, somehow I did not see the other tests.. so you are already testing both versions, allowed and disallowed.

Reply:
> I don't understand. If you don't uncomment them, how would you get the error messages?

Everything is good! I did not saw that you already did what I meant with the `*disallowed*` tests. That makes totally sense now. Somehow I just missed those. However, I meant the error messages that are shown e.g. in the `*disallowed*` tests.. E.g. `TypeError 2527: (153-162): Function declared as pure, but this expression (potentially) reads from the environment or state and thus requires "view".` I was missing somehow the exact reference in the source code.. I was expecting that the thing is somehow written directly in the message. However, I also noticed now that the reference is shown, if `solc` is used:
```
Error: Function declared as pure, but this expression (potentially) reads from the environment or state and thus requires "view".
  --> /Users/alex/git/solidity/test/libsolidity/syntaxTests/test2.sol:68:17:
   |
68 |             pop(chainid())
   |                 ^^^^^^^^^
```

Reply:
If we have e.g. `TypeError 2527: (153-162)`... wouldn't it be nice to extract the corresponding string (eg. string between characters `153-162`).. so we could also read the tests more easily.. e.g. `TypeError 2527: (153-162):  'chainid()' Function declared as pure, but this expression (potentially) reads from the environment or state and thus requires "view"`.

Reply:
Yeah, there were similar ideas in #4620 and #9581. Maybe we should restart the discussion there. @nishant-sachdeva also recently commented on the fact that we don't even have line numbers in tests, which is another similar idea.

The thing is, we don't really have an agreement on what to do so it requires some design discussion.

I think that adding line numbers would probably be uncontroversial, easy to implement and would help a bit so maybe we should create a `good first issue` for that? Not sure if we have enough manpower now to push through a bigger change.

Reply:
Sure, having line numbers would make the navigation in files much more easy. But it will not help during reviews, because the user still need to look at numbers. I just created #12890 that adds the location string to the message. I think that increases the readability of such tests.

Reply:
That's a monster PR :)

I think line numbers still help a bit. You can see them in diffs. With them it's much easier than what I do now when reviewing, i.e. counting errors manually.